pkg/stanza/fileconsumer/metrics.go

```diff
@@ -0,0 +1,125 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package fileconsumer // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer"
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/collector/config/configtelemetry"
+	rcvr "go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+const (
+	scopeName   = "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer"
+	receiverKey = "receiver"
+	nameSep     = "_"
+	typeStr     = "filelog"
+)
+
+var (
+	receiverTagKey    = tag.MustNewKey(receiverKey)
+	statFileReadDelay = stats.Int64("file_read_delay", "the time it takes to read data from a file", stats.UnitMilliseconds)
+)
+
+func init() {
+	_ = view.Register(metricViews()...)
+}
+
+func BuildMetricName(typestr string, metric string) string {
+	return receiverKey + nameSep + typestr + nameSep + metric
+}
+
+// MetricViews returns the metrics views related to fileconsumer.
+func metricViews() []*view.View {
+	receiverTagKeys := []tag.Key{receiverTagKey}
+
+	distributionReadFileDelayView := &view.View{
+		Name:        BuildMetricName(typeStr, statFileReadDelay.Name()),
+		Measure:     statFileReadDelay,
+		Description: statFileReadDelay.Description(),
+		TagKeys:     receiverTagKeys,
+		Aggregation: view.Distribution(10, 25, 50, 75, 100, 250, 500, 750, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000, 20000, 30000, 50000,
+			100_000, 200_000, 300_000, 400_000, 500_000, 600_000, 700_000, 800_000, 900_000,
+			1000_000, 2000_000, 3000_000, 4000_000, 5000_000, 6000_000, 7000_000, 8000_000, 9000_000),
+	}
+
+	return []*view.View{
+		distributionReadFileDelayView,
+	}
+}
+
+type fileConsumerTelemetry struct {
+	level    configtelemetry.Level
+	detailed bool
+	useOtel  bool
```

Comment:
@open-telemetry/collector-approvers, do we expect to support a `useOtel` throughout the collector? Would it be reasonable to assume we're using OTel in new instrumentation at this point?

Reply:
Do we need to discuss this here? I feel that whether to use otel should be unified throughout the project.

Reply:
I'd like a decision. I'll bring it up at Wednesday's SIG

Reply:
Discussed this in SIG today. The consensus from core maintainers is to support only OpenCensus metrics for now. We plan to provide a bridge once https://github.com/open-telemetry/opentelemetry-collector/issues/7454 is complete. Therefore, let's remove the flag.